homeassistant/components/climate/generic_thermostat.py

```diff
@@ -276,7 +276,10 @@ def operation_list(self):
         temperature = kwargs.get(ATTR_TEMPERATURE)
         if temperature is None:
             return
-        self._target_temp = temperature
+        if self._is_away:
```

Comment:
`async_set_temperature` is changing current target temperature, we need maintenance that for all climate devices. Please do not change the behavior of such function.

Reply:
To confirm: your proposed solution to #20255 is to create a separate service to be used in automations (e.g. set_temperature_automated) which will act as my PR above does, not affect the UI elements, be able to handle the conditions described in #20255, and not contravene your architecture rules.

If so, I will look to implement this. However, I would suggest reviewing the current documentation, as generic_thermostat does not operate as the documentation implies.

Reply:
No. I suggest you write a service to change `away_temperature`, so the method should be named `set_away_temperature`. e.g. the `away_temperature` loaded from config at first, then can be modified later on via either automation or services-tool in UI.

Reply:
I'm not sure how many times I can reiterate that **I DO NOT WANT TO CHANGE THE AWAY TEMPERATURE**.

Please re-read my issue #20154. Changing the away temperature _is the exact opposite of what I want_. That's what happens (albeit temporarily) at the moment, when you use automations to run a heating schedule. My PR doesn't even change the away temperature. It changes the _stored temperature_, which is what is used to revert from away mode.

When away mode is enabled, the target temperature shouldn't change via automations. If you have a better way of setting a heating schedule for the generic_thermostat than using automations, please explain what it is.

Reply:
If you want to disallow target temperature change while in away mode, you need fail set_temperature service call, not change its behavior.

I think my original comments are very clear,
> Do not change behavior of async_set_temperature. You can implement a new services to change saved_temperate or away_temperate. Or you can add new parameter to async_set_temperature, since it accepts **kwargs.

My last reply may confuse you, my intention is against you put "automated" into method name, we won't create a service only serve automation.

Regarding the scheduler, there is an open [architecture issue](https://github.com/home-assistant/architecture/issues/148)

Reply:
So your recommendation is a new service called "set_saved_temperature"? Which would require manual logic to be added to an automation to determine whether away_mode is enabled, and to call the correct service? That seems unnecessary and counter-intuitive. 

"automated" was just a name I pulled out of the air. The actual function would act as "set_target_or_saved_temperature_depending_on_away_mode_status". I can look at adding a new service called "set_target_or_saved_temperature" if this would be acceptable.

> you need fail set_temperature service call, not change its behavior.
Can you elaborate on this a bit please? Do you mean something as simple as this?
`if self._is_away:
    return`

Please don't get me wrong. I do want to work with you to find a solution to my issue that meets your approval but doesn't require "workarounds". As it stands, a scheduler that uses the current "set_temperature" doesn't feel like it would work any better than the current setup. I don't believe I can be the only person who this affects.

Reply:
Can I get a response? I raised an issue, and I'm trying to resolve that issue in a way that the entire community can benefit from. Currently I'm running a custom component with my PR and it works fine for me; I'm just trying to feed something back to the project.

Please elaborate on a solution to the issue that you would find acceptable, that DOES NOT talk about changing the away temperature.

You say don't change async_set_temperature, but there are already several different implementation of this function in different climate components, many of which presumably (being actual smart devices themselves) are already capable of handling the issue that this is trying to resolve #20154. Are you saying that a change to async_set_temperature in generic_thermostat.py will affect other thermostats? If so, that's a reasonable objection, and it might have been good to explain that in the first place.